net/http/src/THttpCallArg.cxx

```diff
@@ -140,69 +116,190 @@ TString THttpCallArg::CountHeader(const TString &buf, Int_t number) const
+
+void THttpCallArg::SetXmlContent(std::string &&xml)
+{
+   SetXml();
+   fContent = xml;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set content type as "application/json"
+
+void THttpCallArg::SetJson()
+{
+   SetContentType("application/json");
+}
+
 ////////////////////////////////////////////////////////////////////////////////
-/// set data, posted with the request
+/// Set content type as "application/json" and also assigns content
+/// After method call argument \param json will be in undefined state
+
+void THttpCallArg::SetJsonContent(std::string &&json)
+{
+   SetJson();
+   fContent = json;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set content type as "application/x-binary"
+
+void THttpCallArg::SetBinary()
+{
+   SetContentType("application/x-binary");
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set content type as "application/x-binary" and also assigns content
+/// After method call argument \param bin will be in undefined state
+
+void THttpCallArg::SetBinaryContent(std::string &&bin)
+{
+   SetBinary();
+   fContent = bin;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// \deprecated  Use signature with std::string
+/// Set data, posted with the request
 /// buffer should be allocated with malloc(length+1) call,
 /// while last byte will be set to 0
 /// Than one could use post data as null-terminated string
 
 void THttpCallArg::SetPostData(void *data, Long_t length, Bool_t make_copy)
 {
-   if (fPostData) {
-      free(fPostData);
-      fPostData = nullptr;
-      fPostDataLength = 0;
-   }
-
-   if (length <= 0)
-      return;
+   fPostData.resize(length);
 
-   if (make_copy && data && length) {
-      void *newdata = malloc(length + 1);
-      memcpy(newdata, data, length);
-      data = newdata;
+   if (data && length) {
+      memcpy((void *) fPostData.data(), data, length);
+      if (!make_copy) free(data); // it supposed to get ownership over the buffer
    }
+}
 
-   if (data)
-      *(((char *)data) + length) = 0;
+////////////////////////////////////////////////////////////////////////////////
+/// set data, which is posted with the request
+/// Although std::string is used, not only text data can be assigned -
+/// std::string can contain any sequence of symbols
 
+void THttpCallArg::SetPostData(std::string &&data)
+{
    fPostData = data;
-   fPostDataLength = length;
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// assign websocket handle with HTTP call
+/// takeout websocket handle with HTTP call
+/// can be done only once
 
-void THttpCallArg::SetWSEngine(THttpWSEngine *engine)
+std::shared_ptr<THttpWSEngine> THttpCallArg::TakeWSEngine()
 {
-   if (fWSEngine)
-      delete fWSEngine;
-   fWSEngine = engine;
+   return std::move(fWSEngine);
```

Comment:
Please `reset()` `fWSEngine` afterwards; I'm not sure whether a moved-from fWSEngine is guaranteed to be reset?